projects/core/src/asm/facade/csagent-auth.service.ts

```diff
@@ -0,0 +1,141 @@
+import { Injectable } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { combineLatest, Observable, of } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { CxOAuthService } from '../../auth/user-auth/facade/cx-oauth-service';
+import { UserIdService } from '../../auth/user-auth/facade/user-id.service';
+import { AuthActions } from '../../auth/user-auth/store/actions';
+import {
+  OCC_USER_ID_ANONYMOUS,
+  OCC_USER_ID_CURRENT,
+} from '../../occ/utils/occ-constants';
+import { RoutingService } from '../../routing/facade/routing.service';
+import { UserService } from '../../user/facade/user.service';
+import {
+  AsmAuthStorageService,
+  TokenTarget,
+} from '../services/asm-auth-storage.service';
+import { AsmAuthService } from '../services/asm-auth.service';
+import { AsmActions } from '../store/actions';
+import { StateWithAsm } from '../store/asm-state';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CsAgentAuthService {
+  constructor(
+    protected authService: AsmAuthService,
+    protected authStorageService: AsmAuthStorageService,
+    protected userIdService: UserIdService,
+    protected cxOAuthService: CxOAuthService,
+    protected store: Store<StateWithAsm>,
+    protected userService: UserService,
+    protected routingService: RoutingService
+  ) {}
+
+  /**
+   * Loads a user token for a customer support agent
+   * @param userId
+   * @param password
+   */
+  authorizeCustomerSupportAgent(userId: string, password: string): void {
```

Comment:
IMO would be nicer and cleaner to implement it in `async/await` manner, or switch to using observable flow. 
The same applies to `logoutCustomerSupportAgent`